progress/internal/linear-progress.ts

```diff
@@ -28,16 +29,20 @@ export class LinearProgress extends Progress {
         (this.indeterminate ? 1 : this.value / this.max) * 100
       }%)`,
     };
+
+    const bufferValue = Math.min(this.buffer ?? 0, this.max);
```

Comment:
I'm not sure I entirely follow why we need `Math.min` here to change bufferValue to max. Is it to avoid `bufferValue >= this.max` for `const hideDots`?

Reply:
Yes, so we do not need to worry about that logic with `dotStyles` anymore though I suppose the `hideDots` would take care of it anyway right?

Reply:
Yea I believe so. Do you mind not clamping it and keeping the `bufferValue >= this.max` check?

Semantically "bufferValue" as changed doesn't reflect `this.buffer`, it's really `bufferValueOrMax`. It's a bit confusing to read, and may inadvertently cause problems if future changes don't realize that nuance.

Reply:
@asyncLiz sure I did in the last commit.